openwisp_network_topology/integrations/device/base/models.py

```diff
@@ -185,3 +186,76 @@ def trigger_device_updates(cls, link):
             if 'openwisp_monitoring.device' in settings.INSTALLED_APPS:
                 run_checks = import_string(trigger_device_checks_path)
                 run_checks.delay(device_node.device.pk, recovery=link.status == 'up')
+
+
+class AbstractWifiMesh(UUIDModel):
+    topology = models.OneToOneField(
+        get_model_name('topology', 'Topology'), on_delete=models.CASCADE
+    )
+    ssid = models.CharField(max_length=32, null=False, blank=False)
+
+    class Meta:
+        abstract = True
+
+    @classmethod
+    def create_wifi_mesh_topology(cls, instance, *args, **kwargs):
+        for interface in instance.data.get('interfaces', []):
+            if not interface.get('wireless'):
+                continue
+            if not interface['wireless'].get('mode') in ['802.11s']:
+                continue
+            if not interface['wireless'].get('clients'):
+                continue
+            # This is a mesh interface. Get topology for this mesh.
+            try:
+                mesh_topology = (
+                    cls.objects.select_related('topology')
+                    .get(
+                        ssid=interface['wireless']['ssid'],
+                        topology__organization_id=instance.organization_id,
+                    )
+                    .topology
+                )
+            except cls.DoesNotExist:
+                Topology = load_model('topology', 'Topology')
+                mesh_topology = Topology(
+                    organization=instance.organization,
+                    label=interface['wireless']['ssid'],
+                    parser='netdiff.NetJsonParser',
+                    strategy='receive',
+                    expiration_time=330,
+                )
+                mesh_topology.full_clean()
+                mesh_topology.save()
+                wifi_mesh = cls(
+                    ssid=interface['wireless']['ssid'], topology=mesh_topology
+                )
+                wifi_mesh.full_clean()
+                wifi_mesh.save()
+
+            # Create NetJSONGraph for this mesh
+            graph = {
+                'type': 'NetworkGraph',
+                'protocol': 'OLSR',
+                'version': '0.8',
+                'metric': 'ETX',
```

Comment:
these have to be adjusted, eg: NetJSON, version 1, metric please check if it can be empty